api/client.go

```diff
@@ -22,40 +21,20 @@ type Client struct {
 	Headers http.Header
 }
 
-func checkError(status int, body []byte) error {
-	var apiError Error
+func checkError(resp *http.Response, body []byte) error {
+	if resp.StatusCode >= 200 && resp.StatusCode < 300 {
```

Comment:
True. Generally 3xx codes will be handled by the `http.Client` and never make it here.  If there are too many redirects [defaultCheckRedirect](https://cs.opensource.google/go/go/+/refs/tags/go1.18.1:src/net/http/client.go;l=811-816) returns an error, which will be returned by `client.HTTP.Do`.

The only way to get here from a 3xx status code is a custom `http.Client.CheckRedirect`  function which doesn't error in the same way.

If we don't count 3xx codes as errors, then we'll return `nil` here and in the caller we will try to unmarshal a response, which will fail (because we didn't get a 2xx code, and the user will get a `parsing json response` error.

That seems like the wrong error to return. The user should be informed they ended up with a 3xx status code, so they can fix their proxy settings.

So I think `< 300` is correct, but I can add a comment about why.